File: RetrofitExampleapi/app/src/main/java/com/example/retrofitexampleapi/ApiServices.java
package com.example.retrofitexampleapi;

import com.example.retrofitexampleapi.models.CommentsModels;
import com.example.retrofitexampleapi.models.PostModels;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ApiServices {

    @GET("posts")
    Call<ArrayList<PostModels>>  getpostlist();

    @GET("posts/{id}/comments")
    Call<ArrayList<CommentsModels>> getCommentst(
            @Path("id") int id
    );

    @FormUrlEncoded
    @POST("posts")
    Call<PostModels> getPsotWithField(

            @Field("title") String title,
            @Field("userId") int userId,
            @Field("body") String body
    );

}

File: 4pmclass/RestApiExamples/app/src/main/java/com/example/restapiexamples/ApiServices.kt
package com.example.restapiexamples

import com.example.restapiexamples.models.*
import retrofit2.Call
import retrofit2.http.*

interface ApiServices {
    @GET("posts")
    fun getlist(): Call<MutableList<Postmodels>>

    @GET("posts/{id}/comments")
    fun getcommetlist(
        @Path("id") id:Int
    ): Call<MutableList<Comments>>
    @FormUrlEncoded
    @POST("register.php")
    fun registeryourself(
            @Field("email") email:String,
            @Field("password") password:String,
            @Field("name") name:String
    ):Call<RegisterResponse>


    @GET("getquestion.php")
    fun getquetion(
         @Query("topic") topic:String
    ):Call<Questionlist>


}